### PR TITLE
Add FastFlow model with example and update vision model structure

### PR DESCRIPTION
**Pull request by LukaszMarcjan**
## Summary

Adds the **FastFlow** model. As part of this change, the PaSTe and FastFlow models are refactored onto a shared base, so both reuse the same training, prediction, and thresholding scaffolding.

A new vision model only needs to implement its training logic and a single inference hook (`_inference_maps`) returning per-pixel anomaly maps. A model opts into pixel-level evaluation by returning `score_maps` in its results, with no config flags needed.

## What's included

- **FastFlow model**: architecture, flow blocks, backbone feature extraction, and config.
- **Shared base classes**: PaSTe is rewired onto them, removing the duplicated `fit` / `predict` / thresholding code that FastFlow would otherwise have copied.
- **Shared config layer**: per-model configs only contain model-specific fields.
- **Shared backbone registry**: backbone name lists are deduplicated between models.
- **Scale-invariant pixel thresholding** for F1, Dice, and IoU.
- **Optional reproducibility seed** in the shared config: seeds weight initialization, channel permutations, and data shuffling; `seed=None` keeps the current behavior.

### CODE REVIEW

File: src/pyclad/vision/models/fastflow/fastflow.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            affine_clamping=self.config.affine_clamping,
+        )
+        return FastFlowModule(
+            network=network,
+            learning_rate=self.config.learning_rate,
+            adam_betas=(self.config.adam_beta1, self.config.adam_beta2),
+            weight_decay=self.config.weight_decay,
+        )
+
+    def _inference_maps(self, batch: torch.Tensor) -> torch.Tensor:
```
Comment (Nyderx) [inline src/pyclad/vision/models/fastflow/fastflow.py:38]: _inference_maps permanently leaves the module in eval() mode. It works now because Lightning resets train mode on the next fit(). That's a fragile implicit dependency for a method named for returning maps - let's or restore the prior mode.

Pls. check also the previously added model

**Review by Nyderx — CHANGES_REQUESTED** [review]

**Review by Nyderx — APPROVED** [review]
